streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -661,6 +644,11 @@ private void ensureCopartitioning(Set<String> copartitionGroup,
                     Arrays.sort(topics);
                     throw new TopologyBuilderException(String.format("stream-thread [%s] Topics not co-partitioned: [%s]", streamThread.getName(), Utils.mkString(Arrays.asList(topics), ",")));
                 }
+            } else {
+                if (allRepartitionTopicsNumPartitions.get(topic).numPartitions == NOT_AVAILABLE) {
```

Comment:
Yes. If metadata is missing, we skip over this and hope to get the metadata on the next refresh. But this strategy is not working properly yet. See https://github.com/apache/kafka/pull/2404#discussion_r96803370

We might want to add a check for user-topics if they exist though -- currently, the app would just idle if a user topic does not exits.